compileGroovy: Unable to load class 'io.micronaut.annotation.processing.visitor.JavaClassElement' after upgrade to 3.0.0

### Expected Behavior

After upgrading my micronaut groovy project to version 3.0.0 I get the error message without reference to an error location:
`Task :compileGroovy FAILED

FAILURE: Build failed with an exception.

What went wrong:
Execution failed for task ':compileGroovy'.
io/micronaut/annotation/processing/visitor/JavaClassElement

Unable to load class 'io.micronaut.annotation.processing.visitor.JavaClassElement'`

### Actual Behaviour

_No response_

### Steps To Reproduce

_No response_

### Environment Information

gradle dependencies:

`dependencies {
    implementation("io.micronaut:micronaut-management")
    implementation("io.micronaut:micronaut-validation")
    implementation("io.micronaut.groovy:micronaut-runtime-groovy")
    implementation("io.swagger.core.v3:swagger-annotations")
    implementation("javax.annotation:javax.annotation-api")
    implementation("io.micronaut:micronaut-http-client")
    implementation("io.micronaut:micronaut-tracing")
    implementation("io.micronaut.sql:micronaut-jdbc-hikari")
    implementation("io.micronaut.data:micronaut-data-jdbc")
    implementation("io.micronaut.security:micronaut-security-jwt")
    implementation("io.micronaut.reactor:micronaut-reactor")
    implementation("io.micronaut.reactor:micronaut-reactor-http-client")

    implementation("io.micronaut.rxjava2:micronaut-rxjava2")
    implementation("io.micronaut.rxjava2:micronaut-rxjava2-http-client")
    implementation("io.micronaut.rxjava2:micronaut-rxjava2-http-server-netty")
    
    implementation "jakarta.persistence:jakarta.persistence-api:2.2.2"
    implementation "org.codehaus.groovy:groovy-json"
    implementation "org.codehaus.groovy:groovy-yaml"
    implementation "org.codehaus.groovy:groovy-templates"
    
    implementation group: 'org.apache.poi', name: 'poi-ooxml', version: '5.0.0'
    
    implementation group: 'com.fasterxml.jackson.dataformat', name: 'jackson-dataformat-yaml', version: '2.11.3'
    implementation group: 'com.fasterxml.jackson.datatype', name: 'jackson-datatype-jsr310', version: '2.11.3'
    
    compileOnly("io.micronaut.data:micronaut-data-processor")
    compileOnly("io.micronaut.security:micronaut-security-annotations")
    compileOnly("io.micronaut:micronaut-http-validation")
    compileOnly("io.micronaut.openapi:micronaut-openapi")
    
    runtimeOnly("io.jaegertracing:jaeger-thrift")
    runtimeOnly("ch.qos.logback:logback-classic")
    runtimeOnly("org.postgresql:postgresql")

    testImplementation("org.testcontainers:postgresql")
    testImplementation("org.testcontainers:spock")
    testImplementation("org.testcontainers:testcontainers")
}`

### Example Application

_No response_

### Version

3.0.0

--- a/openapi/src/main/java/io/micronaut/openapi/visitor/AbstractOpenApiVisitor.java
+++ b/openapi/src/main/java/io/micronaut/openapi/visitor/AbstractOpenApiVisitor.java
@@ -25,7 +25,6 @@ import com.fasterxml.jackson.databind.ObjectMapper;
 import com.fasterxml.jackson.databind.PropertyNamingStrategy;
 import com.fasterxml.jackson.databind.annotation.JsonNaming;
 import com.fasterxml.jackson.databind.node.ObjectNode;
-import io.micronaut.annotation.processing.visitor.JavaClassElement;
 import io.micronaut.annotation.processing.visitor.JavaClassElementExt;
 import io.micronaut.core.annotation.AnnotationClassValue;
 import io.micronaut.core.annotation.AnnotationValue;
@@ -49,7 +48,6 @@ import io.micronaut.inject.ast.ElementQuery;
 import io.micronaut.inject.ast.EnumElement;
 import io.micronaut.inject.ast.FieldElement;
 import io.micronaut.inject.ast.MemberElement;
-import io.micronaut.inject.ast.MethodElement;
 import io.micronaut.inject.ast.PropertyElement;
 import io.micronaut.inject.ast.TypedElement;
 import io.micronaut.inject.visitor.VisitorContext;
@@ -825,18 +823,11 @@ abstract class AbstractOpenApiVisitor  {
     }
 
     private boolean doesParamExistsMandatoryInConstructor(Element element, @Nullable Element classElement) {
-        MethodElement methodElement = null;
-        if (classElement instanceof JavaClassElementExt && ((JavaClassElementExt) classElement).getPrimaryConstructor().isPresent()) {
-            methodElement = ((JavaClassElementExt) classElement).getPrimaryConstructor().get();
-        } else if (classElement instanceof JavaClassElement && ((JavaClassElement) classElement).getPrimaryConstructor().isPresent()) {
-            methodElement = ((JavaClassElement) classElement).getPrimaryConstructor().get();
-        }
-
-        if (methodElement != null) {
-            return Arrays.stream(methodElement.getParameters())
-                    .filter(parameterElement -> parameterElement.getName().equals(element.getName()))
-                    .map(parameterElement -> !parameterElement.isNullable())
-                    .findFirst()
+        if (classElement instanceof ClassElement) {
+            return ((ClassElement) classElement).getPrimaryConstructor().flatMap(methodElement -> Arrays.stream(methodElement.getParameters())
+                            .filter(parameterElement -> parameterElement.getName().equals(element.getName()))
+                            .map(parameterElement -> !parameterElement.isNullable())
+                            .findFirst())
                     .orElse(false);
         }
 

